docusaurus/docs/reference/extensions-add-expose.md

```diff
@@ -0,0 +1,99 @@
+---
+id: extensions-add-expose
+title: Understand when to add or expose elements in Extensions
+description: Learn when to add and when to expose en element to an extension point.
+keywords:
+  - grafana
+  - plugins
+  - documentation
+  - API reference
+  - API
+  - extensions
+sidebar_position: 10
+---
+
+Read on to understand when to use `add*` vs `expose*` APIs when developing Extensions in Grafana. This document explains their purpose and provides guidelines on when to use each approach.
+
+## Overview: When to use which API?
+
+Follow these guidelines to create flexible and maintainable UI extensions in Grafana that work seamlessly across plugins.
+
+| Scenario      | Use add* APIs          | Use expose* APIs |
+| -------------- | ------------------ | -------- |
+| Allow external plugins to inject UI elements | ✅ |  ❌  |
+| Add elements to an existing extension point | ✅ |  ❌  |
+| Share a reusable component/function | ❌ |  ✅ |
+| Consume a component/function from another plugin | ❌ |  ✅ |
+
+## Using `add*` APIs
+
+Use the `add*` APIs when you want to enable multiple plugins to extend or contribute to a specific UI area. This allows for dynamic extensibility without direct dependencies between plugins. 
+
+For example:
+
+- Use `add*` when you want to create an extension point and allow other plugins to hook into your UI by adding links, components, or other elements.
+- When contributing to an existing extension point, use `add*` when you want your plugin to add UI elements to an existing extension point in Grafana Core or another plugin.
+
+### Examples
+
+#### Defining an extension point
+
+A plugin defining an extension point where other plugins can contribute components:
```

Comment:
the word `contribute` doesn't trigger my dev mind. I am not really "contributing" anything to other plugins. I am exposing an API  or I am letting other plugins or grafana core render my components but normally as a dev I wouldn't define this as "contribute"